### PR TITLE
terraform: add file and module for terraform integration

### PR DESCRIPTION
**Pull request by xlr-8**
The project is currently using cloudformation, which is a good way to automate things and make it easy for everyone. However some also use terraform to automate their infrastructure, the current PR allows to use both.

In order to help automating the use of autospotting, terraform can be used as a module by other people:
```
> cat example.tf
module "autospotting" {
  source = "github.com/cristim/autospotting/terraform/autospotting"

  autospotting_min_on_demand_number = "${var.asg_min_on_demand_number}"
  autospotting_min_on_demand_percentage = "${var.asg_min_on_demand_percentage}"
  autospotting_regions_enabled = "${var.asg_regions_enabled}"
}
```

This PR allows:
* To use both cloudformation & terraform
* To integrate autospotting in people's infrastructure using terraform (via the module)
* Remove the problem of region from Cloudformation
* Remove the need to have an S3 bucket (the .zip is directly used and uploaded)
* Allow to specify easily variables (from autospotting to lambda customisation)

### CODE REVIEW

**Comment by xlr-8** [comment]
ACK: Need to clean-up markdown, will be done later on.

File: terraform/autospotting.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  autospotting_min_on_demand_number = "${var.asg_min_on_demand_number}"
+  autospotting_min_on_demand_percentage = "${var.asg_min_on_demand_percentage}"
+  autospotting_regions_enabled = "${var.asg_regions_enabled}"
+
+  lambda_zipname = "${var.lambda_zipname}"
+  lambda_runtime = "${var.lambda_runtime}"
+  lambda_memory_size = "${var.lambda_memory_size}"
+  lambda_timeout = "${var.lambda_timeout}"
+  lambda_run_frequency = "${var.lambda_run_frequency}"
+}
```
Comment (cristim) [inline terraform/autospotting.tf]: No new line at the end of file

File: terraform/autospotting/autospotting-policy.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        "iam:PassRole",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
```
Comment (cristim) [inline terraform/autospotting/autospotting-policy.json]: Again

File: terraform/autospotting/lambda-policy.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
```
Comment (cristim) [inline terraform/autospotting/lambda-policy.json]: One more

**Pull request merged**